### ISSUE TITLE
How does the CausalStarGANv2VC model realize streamining inference?

### ISSUE THREAD

**Issue by liusongxiang** [fss1t/CausalStarGANv2-VC#7]
Hi, thank you very much for sharing this great repo.

I am wondering how the causal starganv2 model achieves low-latency real-time voice conversion.
The most confusing part for me is the InstanceNorm2d modules in the causal model. For example, here https://github.com/fss1t/CausalStarGANv2-VC/blob/e1d9bdb86f2d091a43f78f8886e47a58e12eb347/CausalStarGANv2VC/CausalStarGANv2VC/models/generator.py#L114 In my understanding, the layer takes an input feature with shape [N, C, H, W] as input, and computes statistics (i.e., the mean and std) along the axes (H, W). Hence, the model can see all the future frames in the input, thus violating the causal rule.

Please clear me up if my understanding is wrong. Thanks.

**Comment by fss1t** [comment]
Thank you very much for visiting my repository.

When doing streaming inference, we need to fix the normalization parameters using pre-recorded source speaker audio.

```
generator.load_state_dict(checkpoint_path)
generator.train()
_ = generator(source_melspec,target_style_code) # fix norm params by tracking running stats
generator.eval()
# start streaming
```

In this case, it is not strictly any-to-many, but streaming conversion can be done with only one source speaker's utterance.

**Comment by liusongxiang** [comment]
Thanks for your reply.
